File: ios/VectorialData/VectorialData/Core/Models/StockResearch.swift
import Foundation

/// Full research payload for a single pick. Matches
/// `GET /api/picks/research/:ticker`.
///
/// When `locked == true` (non-subscribed users), only the base identification
/// fields + `summaryShort` are populated. Subscribed users get every field.
struct StockResearch: Codable, Equatable, Identifiable {
    let ticker: String
    let name: String
    let sector: String
    let industry: String
    let country: String
    let region: String
    let currency: String
    let summaryShort: String
    /// One human sentence — server-side compacted from `summary_short`,
    /// markdown stripped, enumerations cut. Render verbatim.
    let oneLiner: String?
    let pickNumber: Int?
    let pickDate: String?
    let isSubscribed: Bool
    let locked: Bool

    let summaryWhat: String?
    let summaryWhy: String?
    let summaryRisk: String?
    /// Mom-readable short paragraphs (~280 chars). Server-side compacted
    /// — render verbatim, no client parsing.
    let whyShort: String?
    let riskShort: String?
    /// "LO IMPORTANTE" pills — 3 plain-language vital signs replacing
    /// the old DATOS CLAVE (P/E ratio etc.). Server-built so the UI
    /// stays jargon-free.
    let whatsImportant: [WhatsImportantPill]?
    let peRatio: Double?
    let peForward: Double?
    let dividendYield: Double?
    let marketCapB: Double?
    let analystConsensus: String?
    let analystTarget: Double?
    let analystUpside: Double?
    let waMessage: String?

    var id: String { ticker }
}

struct WhatsImportantPill: Codable, Equatable, Hashable {
    let emoji: String
    let text: String
}

File: android/app/src/main/java/com/vectorialdata/app/core/model/StockResearch.kt
package com.vectorialdata.app.core.model

import kotlinx.serialization.Serializable

/**
 * "Lo importante" pill. `icon` is an SF Symbol name on the wire — Android
 * maps it to a Material icon (with a neutral fallback). `tint` is one of
 * emerald / red / yellow; anything else renders white.
 */
@Serializable
data class WhatsImportantPill(
    val icon: String,
    val tint: String,
    val text: String,
)

/**
 * `GET /api/picks/research/{ticker}`. Mirror of iOS `StockResearch`.
 * When `locked == true` only the base fields + [summaryShort] carry data.
 * `oneLiner`/`whyShort`/`riskShort` are the server-side mom-overrides and
 * must render verbatim (no client compaction).
 */
@Serializable
data class StockResearch(
    val ticker: String,
    val name: String,
    val sector: String,
    val industry: String,
    val country: String,
    val region: String,
    val currency: String,
    val summaryShort: String,
    val oneLiner: String? = null,
    val pickNumber: Int? = null,
    val pickDate: String? = null,
    val isSubscribed: Boolean = false,
    val locked: Boolean = false,
    val summaryWhat: String? = null,
    val summaryWhy: String? = null,
    val summaryRisk: String? = null,
    val whyShort: String? = null,
    val riskShort: String? = null,
    val whatsImportant: List<WhatsImportantPill>? = null,
    // Decoded for parity with iOS but no longer displayed — the UI shows
    // the whats_important pills instead.
    val peRatio: Double? = null,
    val peForward: Double? = null,
    val dividendYield: Double? = null,
    val marketCapB: Double? = null,
    val analystConsensus: String? = null,
    val analystTarget: Double? = null,
    val analystUpside: Double? = null,
    val waMessage: String? = null,
)
